src/components/data/EntitiesContent.js

```diff
@@ -496,6 +500,16 @@ const EntitiesContent = ({
         onDismiss: () => setAddingEntity(false),
         onSuccess: () => setRefreshKey(_.add(1))
       }),
+      addingColumn && h(AddColumnModal, {
+        entityType: entityKey,
+        entityMetadata,
+        workspaceId: { namespace, name },
+        onDismiss: () => setAddingColumn(false),
+        onSuccess: () => {
+          setAddingColumn(false)
+          setRefreshKey(_.add(1))
```

Comment:
Thinking Out Loud: I'm not familiar with the notion of a refresh key. I recognize that this isn't new in this PR, but it's a technique that is relatively new to the codebase and I'm curious what this is for. I'll have to explore this.

Reply:
It's kind of a hack necessitated by having data loading tied to component lifecycle. These "refresh keys" are passed as keys to some component and changing the key causes that component to be unmounted and replaced by a new instance. This causes lifecycle hooks such as `useEffect` and `useOnMount` to re-run.

https://kentcdodds.com/blog/understanding-reacts-key-prop